acme/examples/http01_example.py

```diff
@@ -0,0 +1,271 @@
+    pkey.generate_key(OpenSSL.crypto.TYPE_RSA, CERT_PKEY_BITS)
+    pkey_pem = OpenSSL.crypto.dump_privatekey(OpenSSL.crypto.FILETYPE_PEM,
+                                              pkey)
+    return pkey_pem
+
+
+def new_csr_comp(domain_name, pkey_pem=None):
+    """Create certificate signing request."""
+    if pkey_pem is None:
+        pkey_pem = new_pkey_pem()
+    csr_pem = crypto_util.make_csr(pkey_pem, [domain_name])
+    return pkey_pem, csr_pem
+
+
+def verify_registration(client_acme, regr):
+    """Query registration status."""
+    client_acme.net.account = regr
+    try:
+        return client_acme.query_registration(regr)
+    except errors.Error as err:
+        if err.typ == messages.OLD_ERROR_PREFIX + 'unauthorized' \
+                or err.typ == messages.ERROR_PREFIX + 'unauthorized':
+            logging.info('Status is deactivated')
+        raise
+
+
+def select_http01_chall(orderr):
+    """Extract authorization resource from within order resource."""
+    # Authorization Resource: authz.
+    # This object holds the offered challenges by the server and their status.
+    authz_list = orderr.authorizations
+
+    for authz in authz_list:
+        # Choosing challenge.
+        # authz.body.challenges is a set of ChallengeBody objects.
+        for i in authz.body.challenges:
+            # Find the supported challenge.
+            if isinstance(i.chall, challenges.HTTP01):
+                return i
+
+    raise Exception('HTTP-01 challenge was not offered by the CA server.')
+
+
+@contextmanager
+def challenge_server(http_01_resources):
+    """Manage standalone server set up and shutdown."""
+
+    # Setting up a fake server that binds at PORT and any address.
+    address = ('', PORT)
+    try:
+        servers = standalone.HTTP01DualNetworkedServers(address,
+                                                        http_01_resources)
+        # Start client standalone web server.
+        servers.serve_forever()
+        yield servers
+    finally:
+        # Shutdown client web server and unbind from PORT
+        servers.shutdown_and_server_close()
+
+
+def perform_http01(client_acme, challb, orderr):
+    """Set up standalone webserver and perform HTTP-01 challenge."""
+
+    response, validation = challb.response_and_validation(client_acme.net.key)
+
+    resource = standalone.HTTP01RequestHandler.HTTP01Resource(
+        chall=challb.chall, response=response, validation=validation)
+
+    with challenge_server({resource}):
+
+        # This is the domain used for the local simple_verify. This example
+        # uses a fake domain name in DOMAIN, so this verification would fail
+        # because the DNS would never resolve it to a real IP address. In order
+        # to bypass it and only verify if the standalone server would
+        # successfully reply the correct resource, we use 'localhost' as domain
+        # instead.
+        domain = 'localhost'
+        if not response.simple_verify(challb.chall,
```

Comment:
What's the purpose of this check? We only use it in testing.

Reply:
Yes, it is a local test. Actually it might just confuse people, so I removed.